homeassistant/helpers/script.py

```diff
@@ -1091,11 +1102,9 @@ async def _async_stop_scripts_after_shutdown(hass, point_in_time):
         )
 
 
-async def _async_stop_scripts_at_shutdown(hass, event):
+async def _async_stop_scripts_at_shutdown(hass: HomeAssistant, event: Event) -> None:
     """Stop running Script objects started before shutdown."""
-    async_call_later(
-        hass, _SHUTDOWN_MAX_WAIT, partial(_async_stop_scripts_after_shutdown, hass)
-    )
+    _schedule_stop_scripts_after_shutdown(hass)
```

Comment:
Currently loading any script will trigger this `call_later` for scripts that are started after shutdown.
This causes a lingering timer, with no way to bypass it in tests.

Splitting this up into a separate "schedule" function allows us to patch the function.

If there is a better way to resolve this then I am all ears...